### ISSUE TITLE
could we ack when we receive a new event

### ISSUE THREAD

**Issue by magamal** [rikulo/socket.io-client-dart#379]
`socket.on("new_message",(data)=>{ 

//How to ack this event?
})'`

**Comment by jumperchen** [comment]
Fyi, https://github.com/rikulo/socket.io-client-dart?tab=readme-ov-file#acknowledge-with-the-socket-server-that-an-event-has-been-received

**Comment by magamal** [comment]
Yes I read it before, but if I'm receiving a string from the server how I can ack this event

socket.on("new_message",(data)=>{
 final String messageContent = data;
}

**Comment by steveroseik** [comment]
> Yes I read it before, but if I'm receiving a string from the server how I can ack this event
> 
> socket.on("new_message",(data)=>{ final String messageContent = data; }

Heyy, I had the same issue and figured out the solution. 

It's pretty easy, it's the lack of information in the documentation that makes it hard to figure out.

I'll show you my example:

in my server side I send a nudge to the client, when you send it, the client receives a list of data, consists of the data you sent in the emit and an extra callback function that should be used for acks:

so this is how the data might look like if you used my code below:
['Input the data to be sent.', function()]

`this.server
        .to(opponentId)
        .timeout(5000)
        .emit(
          'nudge',
          'Input the data to be sent.',
          (err, responses) => {
            if (err) {
              /// handle error
            } else {
              // handle responses
            }
          },
        );`



That's why in flutter you'll need to treat the data received as a list, and extract from it both data sent from server and the function;

you can handle the data as you like, and the take the last item as Function and run it as described in their code sample:


`socket.on('nudge', (data) {
    final dataList = data as List;

    final customData = dataList.first;

    final ack = dataList.last as Function;
    ack(null);
});`

**Comment by jumperchen** [comment]
@makamekm Could you provide a reproducible example?

**Comment by makamekm** [comment]
@jumperchen I just fixed it. The error is here:

Solution in one line:
![Image](https://github.com/user-attachments/assets/f054a575-128c-4f11-baae-0059dddb2bf6)
